design/default-global-rate-limit-policy.md

```diff
@@ -0,0 +1,180 @@
+# Default Global RateLimit Policy
+
+Status: Reviewing
+
+## Abstract
+Define a default global rate limit policy in the Contour configuration to be used as a global rate limit policy by all HTTPProxy objects.
+
+## Background
+See https://github.com/projectcontour/contour/issues/5357 for context.
+
+## Goals
+- Define a default global L7 rate limit policy which is used by all HTTPProxy on the `VirtualHost` level.
+
+## Non Goals
+- Default rate limit policy for Route level `rateLimitPolicy`.
+- L4 rate limiting support. 
+
+## High-Level Design
+Currently, the global `rateLimitPolicy` for `VirtualHost` is set per HTTPProxy. This option allows each service to configure its own global `rateLimitPolicy`. However, for a single tenant setup/general usecase, this means you have to edit all the HTTPProxy objects with the same ratelimit policies.
+
+A new field, `DefaultGlobalRateLimitPolicy` (optional), will be added to `RateLimitServiceConfig` which is part of `ContourConfigurationSpec`. This field will define rate limit descriptors that will be added to every `VirtualHost`.
+
+HTTPProxy has to opt-in **explicitly** to use the default global rate limit policy using `rateLimitPolicy.defaultGlobalRateLimitPolicyEnabled` flag and still can have its own global `rateLimitPolicy` which overrides the default one.
+
+### Sample Configurations
+#### contour.yaml
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: contour
+  namespace: projectcontour
+data:
+  contour.yaml: |
+    rateLimitService:
+      extensionService: projectcontour/ratelimit
+      domain: contour
+      failOpen: false
+      defaultGlobalRateLimitPolicy:
```

Comment:
I like the idea of having a flexible solution and it can be useful for sure, but the trade-offs of implementing `APPEND` vs. its benefits might not be balanced. 

Adding `APPEND` means we must maintain many cases for [actions](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-ratelimit-action) that users can set on Envoy. Some of these actions will generate descriptors with custom keys and values, other actions only support setting descriptors value, and Envoy sets the descriptor key. 

So yeah I am not really sure if we want to go this road.